frontend/src/components/UserFlist.vue

```diff
@@ -0,0 +1,152 @@
+                  ></a>
+                  <v-tooltip activator="parent" location="start"
+                    >Download flist</v-tooltip
+                  >
+                </v-btn>
+                <v-btn
+                  @click="copyLink(baseURL + `/` + value)"
+                  class="elevation-0"
+                >
+                  <v-icon icon="mdi-content-copy" color="grey"></v-icon>
+                  <v-tooltip activator="parent">Copy Link</v-tooltip>
+                </v-btn>
+              </template>
+              <template v-else>
+                <span>loading... </span>
+              </template>
+            </template>
+
+            <template #item.last_modified="{ value }">
+              {{ new Date(value * 1000).toString().split("(")[0] }}
+            </template>
+
+            <template v-slot:item.progress="{ value }" class="w-25">
+              <template v-if="value != 100">
+                <v-progress-linear
+                  :model-value="value"
+                  color="#1aa18f"
+                  height="20"
+                  rounded="sm"
+                >
+                  <template v-slot:default="{ value }">
+                    <span class="text-white">{{ Math.floor(value) }}%</span>
+                  </template>
+                </v-progress-linear>
+              </template>
+              <template v-else>
+                <v-chip color="#1aa18f">finished</v-chip>
+              </template>
+            </template>
+          </v-data-table>
+        </v-row>
+      </v-container>
+    </v-main>
+    <Footer />
+  </v-app>
+</template>
+<script setup lang="ts">
+import Navbar from "./Navbar.vue";
+import Footer from "./Footer.vue";
+import { FlistsResponseInterface } from "../types/Flists.ts";
+import { computed } from "vue";
+import { onMounted, ref } from "vue";
+import { toast } from "vue3-toastify";
+import { api, copyLink } from "../client.ts";
+import { filesize } from "filesize";
+
+
+const tableHeader = [
+  { title: "File Name", key: "name" },
+  { title: "Preview", key:"preview"},
+  { title: "Size", key: "size" },
+  { title: "Last Modified", key: "last_modified" },
+  { title: "Download", key: "path_uri", sortable: false },
+  { title: "Progress", key: "progress", width: "20%" },
+];
+const loggedInUser = sessionStorage.getItem("username");
+var flists = ref<FlistsResponseInterface>({});
+const baseURL = import.meta.env.VITE_API_URL;
+let currentUserFlists = computed(() => {
+  return loggedInUser?.length ? flists.value[loggedInUser] : [];
+});
+
+onMounted(async () => {
+  try {
+    flists.value = (await api.get<FlistsResponseInterface>("/v1/api/fl")).data;
+    currentUserFlists = computed(() => {
+      return loggedInUser?.length ? flists.value[loggedInUser] : [];
+    });
+  } catch (error: any) {
+    console.error("Failed to fetch flists", error);
+    toast.error(error.response?.data);
+  }
+});
+</script>
+
+<style>
```

Comment:
It's better to use scoped style
```suggestion
<style scoped>
```